Geek-Collectors-Network/geek-collectors-network
# feat(frontent): add friend request modal

### Summary
Created a "Suggested Friends" list. Users can either delete a user they're not interested in friending, or they can send a friend request; this latter action will open a modal for the user to type in a custom message.

Created a pending request list. This will allow a user to accept or ignore a friend request.

- Created a `GeneralModal` component
- Create a specific `FriendRequestModal` component
- Created a `UserList` page that has two tabs: one for pending friend requests, one for suggested friends
- Created a test page to see the list in action

Note: the `UserList` page will need refactoring.
Note: on the "Friend Suggestion" page of `UserList`, the "+" and "x" are of different thickness and will need updating

### Testing
Note: Cannot use deploy link, because the "userlist" must be navigated to manually, via the address bar

1. Navigate to `frontend` folder; open a terminal at this location
2. Type in `npm run start:dev`
3. Open Chrome and navigate to `http://localhost:8080/userlist`
4. You should open up on the "Pending Requests" tab
5. Click the checkmark on a user; a console message should appear, and the user should disappear from the list
6. Click the "x" on a user; a console message should appear, and the user should disappear from the list
7. Remove all uses from the list; "No pending requests" should appear
8. Navigate to the "Friend Suggestions" tab
9. Click the "x";  a console message should appear, and the user should disappear from the list
10. Click the checkmark; a modal should pop up. Type in a message and click "Send"; console messages should appear and the user should disappear from the list.
11. Click the checkmark again. Click on the "Close" button, on the "x" button at the top, and outside the modal; all three should close the modal, print no console messages, and the user should remain on the list.
12. Remove all users; "No suggested friends" should appear
